[HOTFIX] Added and fixed specifications in class PermissionsHandler

francescobau opened the pull request:
- Added @ NonNull and @ Nullable
- Added specifications for Context parameters
- Changed specifications in method checkPermissions.
- Added check if permissions[] is null

francescobau committed:
[HOTFIX] Added and fixed specifications

- Added @ NonNull and @ Nullable
- Added specifications for Context parameters
- Changed specifications in method checkPermissions.

AlbertoUrsino commented on app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java:
```diff
@@ -55,11 +62,12 @@ public static boolean checkPermissions(Context context, @NonNull String[] permis
      * @author Francesco Bau'
      */
-    private static boolean isGranted(Context context, String permission) {
+    private static boolean isGranted(@NonNull Context context, @NonNull String permission) {
```
This method is called only inside this class, in particular after you already check if `String[] permissions` is null or not.
For example in line 53 and 32.
After saying that, you can be sure that this variable can NEVER be null.

francescobau replied:
I see.
Thank you for the suggestion.

francescobau replied:
Thank you for the suggestion.
But I've put a @ NonNull annotation because, if this class updates, and another method tries to pass a null String, isGranted would throw an exception, because of method checkSelfPermission(@ NonNull String), and it would be hard to understand immediately, or to test it out, since it's a private method...

I thought this way. Am I wrong? What do you think about it?

francescobau replied:
In doubt, I'm gonna change something... You'll see it in next commit.

francescobau replied:
I changed it, now it's called `isNotGranted` . What do you think about it?

AlbertoUrsino commented on app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java:
```diff
@@ -55,11 +62,12 @@ public static boolean checkPermissions(Context context, @NonNull String[] permis
      *
-     * @param permission The permission to check.
+     * @param context    The target Context. It can't be null.
+     * @param permission The permission to check. It can't be null.
```
```suggestion
     * @param permission The permission to check.
```

AlbertoUrsino commented on app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java:
```diff
@@ -20,33 +21,39 @@
-                return false;
+    public static boolean checkPermissions(@NonNull Context context, @Nullable String[] permissions) {
+        if (permissions != null) {
+            for (String permission : permissions) {
```
Inside this method you use the "for-each" type of for statement.

francescobau replied:
Aw sheit...
Thank you for spotting that.

francescobau replied:
Yes, because it checks if all permissions are granted.
And, if String[] permissions is null, it means there's nothing to check.

francescobau replied:
Oh, now I see what do you mean...
Now I've always used a foreach statement.

AlbertoUrsino commented on app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java:
```diff
@@ -20,33 +21,39 @@
-                deniedPermissions.add(permissions[i]);
-                arrayLength++;
+        if (permissions != null) {
+            for (int i = 0; i < permissions.length; i++) {
```
Inside this method you use the classic for loop statement.
Use always the same!

francescobau replied:
Oh, now I see what you mean. Thank you.
I'm gonna fix that right now.

albertoursino reviewed (changes requested):
Good code all in all but I suggested to change some things.

francescobau committed:
Changed method isGranted, always used foreach loop in PermissionsHandler.

> Method isGranted is now called isNotGranted
> Changed specifications of method isNotGranted
> isGranted accepts a null String, and, since someone is asking if (nothing) is not-granted or not, it will return false.

francescobau commented:
I think I improved the code now...
I also realized that the method isGranted was supposed to be called isNotGranted, instead...
So I've been editing something in there, as well...

francescobau committed:
Updated @ author
